cleanupsdemo/cleanups_demo.go

```diff
@@ -13,6 +13,8 @@ func main() {
 	const cycles = 100
 	const batchSize = 100
 
+	ctx := context.Background()
+
```

Comment:
did we need to do this to demonstrate the newly-discovered leak? I wouldn't have expected it to make a difference since this context isn't cancelable

Reply:
i did this to toy around with it. it actually make a difference whether you leave this as `Background()` or a `WithCancel()` of such in this case since we're still referring to the contexts in the "leaked dependent waiters", it's just a slightly clearer spelling i felt.

you can take this file and run it on the main branch and see that 400 goroutines aren't halted before this PR :+1: